main.js

```diff
@@ -0,0 +1,309 @@
+/* jshint -W097 */
+/* jshint strict: false */
+/* jslint node: true */
+'use strict';
+
+const utils = require('@iobroker/adapter-core');
+const objectHelper = require('@apollon/iobroker-tools').objectHelper; // Get common adapter utils
+const mqttServer = require(__dirname + '/lib/mqtt');
+const coapServer = require(__dirname + '/lib/coap');
+const adapterName = require('./package.json').name.split('.').pop();
+const tcpPing = require('tcp-ping');
+const events = require('events');
+const eventEmitter = new events.EventEmitter();
+
+class Shelly extends utils.Adapter {
+
+    constructor(options) {
+        super({
+            ...options,
+            name: adapterName,
+        });
+
+        this.serverMqtt = null;
+        this.serverCoap = null;
+        this.pollTimeout = null;
+
+        this.onlineDevices = {};
+
+        this.on('ready', this.onReady.bind(this));
+        this.on('stateChange', this.onStateChange.bind(this));
+        this.on('message', this.onMessage.bind(this));
+        this.on('unload', this.onUnload.bind(this));
+    }
+
+    async onReady() {
+        try {
+            this.log.info('Starting Adapter ' + this.namespace + ' in version ' + this.version);
+
+            /*
+            this.config.polltime = Number(this.config.polltime) || 0;
+            if (this.config.polltime > 0) {
+                this.log.info('Polltime of the shelly devices: ' + this.config.polltime + ' sec.');
+            } else {
+                this.log.info('Polltime of the shelly devices: ' + this.config.polltime + ' sec.');
+            }
+            */
+
+            // Check sentry configuration
+            if (await this.setSentryLogging(this.config.sentry_enable)) return;
+
+            // Upgrade older configs
+            await this.migrateConfig();
```

Comment:
Best return a boolean here when you migrated something to do not continue with the logic ... else stuff gets iitialized just to be killed "in parallel" because of the object change